plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/NewApplicationWizardModel.java

```diff
@@ -196,11 +217,74 @@ public void loadResources() {
 
 	@Override
 	public void setParameters(List<IParameter> parameters) {
-		firePropertyChange(PROPERTY_PARAMETERS, this.parameters, this.parameters = parameters);
-		this.originalValueMap = new HashMap<String, String>(parameters.size());
-		for (IParameter param : parameters) {
-			originalValueMap.put(param.getName(), param.getValue());
+		firePropertyChange(PROPERTY_PARAMETERS, this.parameters, this.parameters = injectProjectParameters(getEclipseProject(), parameters));
+		originalValueMap = parameters.stream().collect(Collectors.toMap(IParameter::getName,IParameter::getValue));
+	}
+
+	private static List<IParameter> injectProjectParameters(org.eclipse.core.resources.IProject project, List<IParameter> originalParameters) {
+		if (originalParameters == null || originalParameters.isEmpty()) {
+			return originalParameters;
+		}
+		Map<String, String> projectParams = getProjectParameters(project);
+
+		List<IParameter> newParameters = originalParameters.stream().map(p -> { 
+			IParameter clone = p.clone();
+			String value = projectParams.get(clone.getName());
+			if (value != null) {
+				clone.setValue(value);
+			}
+			return clone;
+		}).collect(Collectors.toList());
+
+		return newParameters;
+	}
+
+	private static Map<String, String> getProjectParameters(org.eclipse.core.resources.IProject project) {
+		if(project == null) {
+			return Collections.emptyMap();
+		}
+		Map<String,String> projectParams = new HashMap<>();
+		String gitRepo = getDefaultRemoteRepo(project);
+		if (gitRepo != null) {
+			projectParams.put("SOURCE_REPOSITORY_URL", gitRepo);
+			projectParams.put("GIT_URI", gitRepo);//legacy key
+			
+			String branch = getDefaultBranch(project);
+			projectParams.put("SOURCE_REPOSITORY_REF", branch);
+			projectParams.put("GIT_REF", branch);//legacy key
+			
+			//Setting the context dir is a really bad idea if we're dealing with a multi module project
+			//Better let the user do it manually if needed.
+			//String contextDir = getDefaultContextDir(project);
+			String contextDir = "";
+			projectParams.put("CONTEXT_DIR", contextDir);
+			projectParams.put("GIT_CONTEXT_DIR", contextDir);//legacy key
 		}
+		return projectParams;
+	}
+
+	private static String getDefaultBranch(org.eclipse.core.resources.IProject project) {
```

Comment:
Potentially yes, there's still the question of Exception handling though. most EGitUtils methods throw core exceptions so we'd still need to wrap in a try catch block in the UI layer